### ISSUE TITLE
Bug: Some Areas Fail to Turn on Lights (But Presence is Detected) - Root Cause Found

### ISSUE THREAD

**Issue by 503Dev** [jseidl/hass-magic_areas#296]
**Describe the bug**
After setting up Magic Areas, some areas do not turn on lights as expected.  These areas may turn off the lights, but not back on.  Analyzing their Area entity, it can be seen they detect the presence correctly, but no lights are turned on.

**To Reproduce**
1. Configure the area with Magic Areas
2. Enable the "area aware media player" and lights
3. Do not select a valid media player / select none in the popup 
4. Test the functionality

**Expected behavior**
An error stating a media player must be selected, automatic turning off of the feature if none is selected, etc.

**Additional context**
What actually happens is that Magic Areas allows you to select this feature, but not select any media player.  No error or issue is thrown during configuration.  There are many reasons one might do this, accidentally clicking next, enabling it but then planning to move a media player later and update or the media player becoming invalid (unavailable).

**Solution**
If this happens to you, then just reconfigure the area and disable the area aware media player feature or select a valid media player.  I tested this one two installs, and it reproduces every time, I almost ditched Magic Areas thinking it was broken when really it's just a weird bug (or rather maybe a GUI oversight).

**Logs**
The only place there is any hint of this logged is in Settings -> System -> Logs -> HA Core.  You will notice an error regarding "null" but it's not specific enough to immediately tip you off.
![image](https://github.com/jseidl/hass-magic_areas/assets/2231329/bd3a2363-8661-47eb-b57b-a3a2fed851ec)

**Comment by ethank** [comment]
I'm seeing really inconsistent behavior on turning lights on with Magic Area. The concept is great, but it's unpredictable if it works. I've seen this error, and the above fixed it, but some areas the lights still won't activate.

**Comment by thargy** [comment]
I am getting those `Null data[]` errors when creating new configurations, no matter what values I place in the fields.  I've set everything up and no lights are switching on/off, even though I can see the occupancy sensor trigger, and the secondary state includes 'dark'.

**Comment by jseidl** [comment]
Sorry this is a debug "warning" that I've left behind. Shouldn't cause any issues. 

For people with light control not working, magic areas creates a switch called "Light Control (Area Name)". Make sure it's on!

**Comment by 503Dev** [comment]
> Sorry this is a debug "warning" that I've left behind. Shouldn't cause any issues.
> 
> For people with light control not working, magic areas creates a switch called "Light Control (Area Name)". Make sure it's on!

Sorry? Don't be. Your work is amazing. Thank you! Debug messages are appreciate as a fellow software dev. Now that you've mentioned that - it's interesting but the issue I mentioned here is very valid even if the warning is unrelated. I can reproduce it 100%. So it seems important for everyone to be aware that enabling the media player group / smart aware and not adding an actual media player will break that area. 

I always leave the area switches on but I think those are off by default, good reminder. I've been using Magic Areas happily since. I do get some wonky behaviour with presence hold sometimes and I'm troubleshooting that. Randomly if an area is dark and I hit presence hold it won't turn on the lights. It will keep them on if I turn them on but it won't turn them on. Sometimes it does work just fine though. It's probably some dumb config issue I did.

**Comment by jseidl** [comment]
Thanks for the nice words! Can you cut a separate bug for the media player issue? I don't want it to get lost.

For the Light Groups, sigh, I need to make it simpler to understand, it is a bit convoluted with the area states and all. Make sure you read the wiki on it https://github.com/jseidl/hass-magic_areas/wiki/Light-Groups. 

tl;dr is, Light Group by default, it's only a group. The "magic" functionality you have to turn on that switch I've mentioned (Light Control (Area Name)). It is off by default to prevent driving people crazy but at the same time confuses people who are looking for the "magic" action.

The Light Groups can be tied to given area states (on the Config it's called "States which XYZ lights are turned on") which means they will only turn on if the area is in that given state and will turn off if the area leaves that state. 

There's a catch to that in the other configuration "When should overhead lights be controlled?" which the options are "occupancy" or "state". The "state" option is what you would expect, the state changes, stuff happen. "occupancy" means that lights will be only turned on when the area first goes from "clear > occupied" and will only turn off when clear.

Let me know if something on the wiki isn't clear so I can update.

**Comment by jseidl** [comment]
Ok so this issue should be fixed by #300, if it still persists please reopen
